src/main/java/com/google/api/codegen/transformer/PageStreamingTransformer.java

```diff
@@ -64,6 +64,17 @@
     return descriptors;
   }
 
+  public List<PageStreamingDescriptorClassView> generateDescriptorClasses(
+      DiscoGapicInterfaceContext context) {
+    List<PageStreamingDescriptorClassView> descriptors = new ArrayList<>();
+
+    for (com.google.api.codegen.discovery.Method method : context.getPageStreamingMethods()) {
```

Comment:
It looks like if you had an abstraction of `Method` you could avoid duplicating at least this method, and maybe the rest of this file too.